### ISSUE TITLE
Add constant folding analyser

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/BenchmarkDotNet#731]
If the result == 0 then most probably the benchmark was returning a const value. We should warn the users about it.

Example:

```cs
[Config(typeof(JitsConfig))]
public class MathBenchmarks
{
    private class JitsConfig : ManualConfig
    {
        public JitsConfig()
        {
            Add(Job.Default.With(Jit.LegacyJit).With(Platform.X86).WithId("Legacy x86"));
            Add(Job.Default.With(Jit.LegacyJit).With(Platform.X64).WithId("Legacy x64"));
            Add(Job.Default.With(Jit.RyuJit).With(Platform.X64).WithId("Ryu x64"));
        }
    }

    [Benchmark]
    public double Sqrt14() 
        => Math.Sqrt(1) + Math.Sqrt(2) + Math.Sqrt(3) + Math.Sqrt(4) +
           Math.Sqrt(5) + Math.Sqrt(6) + Math.Sqrt(7) + Math.Sqrt(8) +
           Math.Sqrt(9) + Math.Sqrt(10) + Math.Sqrt(11) + Math.Sqrt(12) +
           Math.Sqrt(13) + Math.Sqrt(14);
}
```

![image](https://user-images.githubusercontent.com/6011991/39320000-a69de034-4937-11e8-891a-ba2381482dc6.png)

**Comment by Rizzen** [comment]
I'll take a look.

**Comment by adamsitnik** [comment]
Fixed in #906
